tests/test_waku_lightpush.nim

```diff
@@ -115,4 +117,36 @@ suite "Waku Lightpush":
       requestError == error
 
     ## Cleanup
-    await allFutures(clientSwitch.stop(), serverSwitch.stop())
\ No newline at end of file
+    await allFutures(clientSwitch.stop(), serverSwitch.stop())
+
+
+  asyncTest "incorrectly encoded request should return an erring response":
+    ## Setup
+    let
+      serverSwitch = newTestSwitch()
+      handler = proc(peer: PeerId, pubsubTopic: PubsubTopic, message: WakuMessage): Future[WakuLightPushResult[void]] {.async.} =
+        ## this handler will never be called: request must fail earlier
+        return ok()
+      server = await newTestWakuLightpushNode(serverSwitch, handler)
+
+    ## Given
+    let
+      fakeBuffer = @[byte(42)]
+      fakePeerId = PeerId.init(PrivateKey.random(ECDSA, (newRng())[]).tryGet()).tryGet()
+    
+    ## When
+    let
+      pushRpcResponse = await server.handleRequest(fakePeerId, fakeBuffer)
+      requestId = pushRpcResponse.requestId
+    
+    ## Then
+    check:
+      requestId == ""
+      pushRpcResponse.response.isSome()
```

Comment:
Shall we also check the received message response in this case?

Reply:
In the following lines, we check that `isSuccess` is false, and that `info` is (more precisely, after the suggested change above, contains)  `decodeRpcFailure`. What else do you think we should check here?

Reply:
ah ok! I missed that. I was kind of expecting a hardcoded value. Didn't notice you were properly using that `const`.